spec/mongoid/attributes/nested_spec.rb

```diff
@@ -4249,6 +4249,43 @@ class BandWithAllowDestroyedRecords < Band
           end
         end
       end
+
+      context "when the relationship is has many" do
+
+        context "when a nested attribute is set and allow_destroy is true, and cache counter and touch are true in the sub-document" do
+
+          let(:party) do
+            party = Party.create(name: 'Ruby Conference', guests_attributes: [{ name: 'attendee 1'}])
+            party.reload.guests
+            party.guests.to_a
+            party.update_attributes(name: 'Ruby Conference', guests_attributes: [{ id: party.guests.first, name: 'attendee 1', _destroy: '1'}])
+            party.reload.guests
+            party.guests.to_a
+            party
+          end
+
+          let(:party_timestamps) do
+            party = Party.create(name: 'Ruby Conference', guests_attributes: [{ name: 'attendee 1'}])
+            party.reload.guests
+            party.guests.to_a
+            time_one = party.updated_at
+            sleep 2
```

Comment:
You can use [travel helpers from ActiveSupport](http://api.rubyonrails.org/classes/ActiveSupport/Testing/TimeHelpers.html) instead of sleep to reduce time overhead.